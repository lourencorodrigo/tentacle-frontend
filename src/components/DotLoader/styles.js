import styled, { keyframes, css } from 'styled-components';

const bouncedelay = keyframes`
  0%, 80%, 100% {
    transform: scale(0);
  } 40% {
    transform: scale(1.0);
  }
`;

const animation = css`
  animation: ${bouncedelay} 1.4s infinite ease-in-out both;
`;

export const Spinner = styled.div`
  text-align: center;

  & > div {
    width: 10px;
    height: 10px;
    background-color: white;
    border-radius: 100%;
    display: inline-block;
  }
`;

export const Bounce1 = styled.div`
  ${animation}
  animation-delay: -0.32s;
`;

export const Bounce2 = styled.div`
  ${animation}
  animation-delay: -0.16s;
`;

export const Bounce3 = styled.div`
  ${animation}
`;
