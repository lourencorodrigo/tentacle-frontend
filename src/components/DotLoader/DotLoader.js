import React from 'react';

import { Spinner, Bounce1, Bounce2, Bounce3 } from './styles';

const DotLoader = () => (
  <Spinner>
    <Bounce1 />
    <Bounce2 />
    <Bounce3 />
  </Spinner>
);

export default DotLoader;
