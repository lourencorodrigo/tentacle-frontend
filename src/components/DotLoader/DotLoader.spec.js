import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';

import DotLoader from './';

describe('dotloader component', () => {
  const dotloader = <DotLoader />;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(dotloader, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should compare the component with a snapshot', () => {
    const component = ReactTestRenderer.create(dotloader);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
