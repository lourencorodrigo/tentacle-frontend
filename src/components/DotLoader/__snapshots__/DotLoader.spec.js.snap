// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotloader component should compare the component with a snapshot 1`] = `
<div
  className="sc-bdVaJa gQrzl"
>
  <div
    className="sc-bwzfXH kWVtkz"
  />
  <div
    className="sc-htpNat SwUlm"
  />
  <div
    className="sc-bxivhb huGgTa"
  />
</div>
`;
